src/www/ui/api/index.php

```diff
@@ -95,7 +109,16 @@
 AppFactory::setContainer($container);
 AppFactory::setResponseFactory(new ResponseFactoryHelper());
 $app = AppFactory::create();
-$app->setBasePath(BASE_PATH);
+$app->setBasePath($BASE_PATH);
+
+// Custom middleware to set the API version as a request attribute
+$apiVersionMiddleware = function (Request $request, $handler) use ($apiVersion) {
+  $request = $request->withAttribute('apiVersion', $apiVersion);
+  return $handler->handle($request);
+};
+
+// Attach the middleware globally to be applied to all routes
+$app->add($apiVersionMiddleware);
```

Comment:
Please move the middleware addition after the comment bellow (just before the `$dbConnected` check). And also update the documentation in the comment to write about this new middleware.